src/poly.rs

```diff
@@ -210,12 +208,14 @@ impl<B: Borrow<Self>> ops::MulAssign<B> for Poly {
 
 impl ops::MulAssign<Fr> for Poly {
     fn mul_assign(&mut self, rhs: Fr) {
-        if rhs.is_zero() {
-            self.zeroize();
-            self.coeff.clear();
+        if bool::from(rhs.is_zero()) {
+            //TODO IC review this
+            //self.zeroize();
+            //self.coeff.clear();
+            *self = Poly::zero();
```

Comment:
I think the old code should still work here, if you keep the old logic, we'll be able to just change the `Zeroize` logic to properly zero-out the polynomial secrets once we have that in `blstrs`.